receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -253,10 +262,10 @@ func (jr *jReceiver) stopTraceReceptionLocked() error {
 }
 
 const collectorReceiverTagValue = "jaeger-collector"
+const tchannelCollectorReceiverTagValue = "jaeger-tchannel-collector"
 
-func (jr *jReceiver) SubmitBatches(ctx thrift.Context, batches []*jaeger.Batch) ([]*jaeger.BatchSubmitResponse, error) {
+func consumeTraceData(ctx context.Context, batches []*jaeger.Batch, consumer consumer.TraceConsumer) ([]*jaeger.BatchSubmitResponse, error) {
```

Comment:
A stab in the dark but are there any implications of passing an interface value as an argument on hot path? Will accepting an interface value and calling a method on it incur some perf penalty significant enough for the hot path? I don't know if it is something to worry about but wanted to point it out in case someone has an opinion. Assuming it does, would resolving the method to a function type and then passing a callback be any better at all?

Reply:
Hi @owais, thanks for pointing that out.
However I'm not sure I'm clear about the fix? If I understand correctly a dynamic dispatch for the interface will be made regardless of this change? Please correct me if I'm wrong here.

Reply:
I think you are right. Just something that caught my eye and I wasn't sure about it. I was thinking that having the object on a struct would resolve the "real" type only once but I suppose that won't happen anyway unless we convert  the interface value to a concrete type so it's all probably the same. Outside the scope of this PR, it would be interesting to benchmark interface vs concrete type method calls on the hot path and see if it is even significant enough.

Reply:
> it would be interesting to benchmark interface vs concrete type method calls on the hot path and see if it is even significant enough.

Yeah, that sounds a reasonable approach here. What occurred to me was to pass the function to be called directly.